### PR TITLE
Removed CNMEM files, removed CMake switch

### PR DESCRIPTION
**Pull request by borisfom**
Removing CNMEM switches
Removed CNMEM references

### CODE REVIEW

File: src/caffe/util/gpu_memory.cpp [hunk 1]
```diff
@@ -142,21 +120,14 @@ namespace caffe {
       // find out the smallest GPU size
       if (poolsize_ == 0 || poolsize_ > free_mem)
         poolsize_ = free_mem;
-#if USE_CNMEM
       devs[i].device = gpus[i];
       devs[i].size = free_mem;
       devs[i].numStreams = 0;
       devs[i].streams = NULL;
-#endif
```
Comment (lukeyeager) [inline src/caffe/util/gpu_memory.cpp]: Don't you want to delete all these lines about `devs[i]`?

File: src/caffe/util/gpu_memory.cpp [hunk 2]
```diff
@@ -177,15 +148,11 @@ namespace caffe {
       }
 
     CUDA_CHECK(cudaSetDevice(initial_device));
-#if USE_CNMEM
     delete [] devs;
-#endif
```
Comment (lukeyeager) [inline src/caffe/util/gpu_memory.cpp]: Don't you want to delete this line about `devs`?

Comment (borisfom) [inline src/caffe/util/gpu_memory.cpp]: Yes! Sorry my local cuda was a mess at the moment so I only compiled with cpu and left this to Travis. Can you do it or wait another hour ?

Sent using OWA for iPhone
________________________________

**Comment by lukeyeager** [comment]
@borisfom I'll clean up this PR and merge it myself since you're busy with other things.

File: src/caffe/util/gpu_memory.cpp (hunk 2 above)
Comment (borisfom) [inline src/caffe/util/gpu_memory.cpp]: Actually, never mind - I just fixed it.

From: Boris Fomitchev <bfomitchev@nvidia.com<mailto:bfomitchev@nvidia.com>>
Date: Friday, November 20, 2015 at 2:05 PM
To: NVIDIA/caffe <caffe@noreply.github.com<mailto:caffe@noreply.github.com>>, NVIDIA/caffe <reply@reply.github.com<mailto:reply@reply.github.com>>
Subject: Re: [caffe] Removed CNMEM files, removed CMake switch (#81)

Yes! Sorry my local cuda was a mess at the moment so I only compiled with cpu and left this to Travis. Can you do it or wait another hour ?

Sent using OWA for iPhone
________________________________

**Comment by lukeyeager** [comment]
Nevermind, it looks like you already re-created this at #82.

**Pull request closed without merging**